CONTRIBUTING.md

```diff
@@ -39,19 +39,39 @@ This subfolder contains some example `.rst` files that show how to implement spe
 ------
 ## Releases
 
-1. Bump the version:
-   1. `git pull main`
-   2. Create a new branch
-   3. Bump `setup.py` and `qiskit_sphinx_theme/__init__.py` to use the new version, e.g. https://github.com/Qiskit/qiskit_sphinx_theme/pull/207
-   4. PR the change and land it
-2. Push the Git tag:
-   1. `git pull main` to pull the version bump. If other commits have landed since the version bump, use `git revert --hard <sha>` to change to the version bump's commit (you can find the SHA with `git log`).
-   2. `git tag <version>`, e.g. 1.11.0
-   3. `git push upstream <version>`
-3. Check that the release worked:
+We use [semantic versioning](https://semver.org/). When starting a new minor release series like `1.11`, we create a new Git branch `1.11` and then selectively cherry-pick any bug fixes from `main` into that release.
+
+So, the release process changes whether you are starting a new minor release series (like `1.11` or `1.12`) vs. releasing later versions of the series. The first release in a new series will be `1.x.0rc1`, e.g. `1.11.0rc1`.
```

Comment:
I think I'm struggling with the term "first release in a series" I think this quite vague and would prefer if we could talk about "releases" and "release candidates" which is more widely used terminology in the industry.